packages/front/src/SingleCardPokemon.tsx

```diff
@@ -0,0 +1,250 @@
+        this.handleAddToCart = this.handleAddToCart.bind(this);
+        this.handleGoToFav = this.handleGoToFav.bind(this)
+    }  
+
+    public getSingleCard() {
+        getSingleCardPokemon(this.props.match.params.indexPokemon).then(resp => {
+                this.setState({
+                    pokemon: resp.card
+                })
+            })
+    }
+
+    public  printIcon(cost:any){
+        switch(cost) {
+            case 'Colorless':
+                return <span className="singlePoke--atackImg singlePoke--iconColorless"/>;
+            case 'Grass':
+                return <span className="singlePoke--atackImg singlePoke--iconGrass"/>;
+            case 'Fire':
+                return <span className="singlePoke--atackImg singlePoke--iconFire"/>;
+            case 'Water':
+                return <span className="singlePoke--atackImg singlePoke--iconWater"/>;
+            case 'Fighting':
+                return <span className="singlePoke--atackImg singlePoke--iconFight"/>;
+            case 'Lightning':
+                return <span className="singlePoke--atackImg singlePoke--iconLightning"/>;
+            case 'Psychic':
+                return <span className="singlePoke--atackImg singlePoke--iconPsychic"/>;
+            case 'Fairy':
+                return <span className="singlePoke--atackImg singlePoke--iconFairy"/>;
+            case 'Metal':
+                return <span className="singlePoke--atackImg singlePoke--iconMetal"/>;
+            case 'Darkness':
+                return <span className="singlePoke--atackImg singlePoke--iconDarkness"/>;
+            case 'Dragon':
+                return <span className="singlePoke--atackImg singlePoke--iconDragon"/>;
+            default:
+                return <span>N/A</span>
+        }
+    }
+    public handleAddToCart(){
+        this.setState((state:any) => {
+            const cartItemsId = state.pokemon.id;
+            const storageId = localStorage.getItem('cartItems');
+            if(cartItemsId !== storageId ){
+                localStorage.setItem('cartItems', JSON.stringify(cartItemsId));
+            } else {
+                console.log("id już istnieje")
+            }
+            return cartItemsId
+        }) 
+    }
+
+    public handleGoToFav(){
+        this.props.history.push(`${PageEnum.FAVOURITE}`);
+    }
+
+    componentDidMount(){
+        this.getSingleCard()
+    }
+
+    render(){
+        const {pokemon} = this.state
+        return(
+            <div className="singlePoke">
+                <img className="singlePoke--imgCard" src={`${pokemon.imageUrl}`} alt=""/>
+                <Card className="singlePoke--cardBox">
+                    <Card.Body>
+                        <div className="singlePoke--buttonStyle">
+                            <button className="btn btn-primary singlePoke--marginRight pokemonCard--buttonHeight" onClick={(e)=>this.handleAddToCart()}>Dodaj do ulubionych</button>
```

Comment:
niepotrzebne funkcje strzałkowe - można to zrobić bez generowania funkcji w renderze

Reply:
Zabezpiecz baton by był disabled jesli element jest już w ulubionych